language/lang.go

```diff
@@ -129,6 +129,21 @@ type GenerateArgs struct {
 	OtherEmpty, OtherGen []*rule.Rule
 }
 
+// HasDefaultVisibility returns whether oldFile contains a "package" rule with
+// a "default_visibility" attribute. Rules generated by Gazelle should not
+// have their own visibility attributes if this is the case.
+func (args GenerateArgs) HasDefaultVisibility() bool {
```

Comment:
I think this should be an exported method of `rule.File` instead. That would make it more generally useful outside of the `Generate` method. I could imagine it being used in the `Resolve` method for extensions that want to consider visibility.